packages/codegen-ui-react/lib/react-studio-template-renderer-helper.ts

```diff
@@ -63,33 +62,37 @@ export function transpile(
 
     const componentText = prettier.format(transpiledCode, { parser: 'typescript', plugins: [parserTypescript] });
 
-    /* createProgram is less performant than traspileModule and should only be used when necessary.
+    /*
+     * createProgram is less performant than traspileModule and should only be used when necessary.
      * createProgram is used here becuase transpileModule cannot produce type declarations.
+     * We execute in a virtual filesystem to ensure we have no dependencies on platform fs in this stage.
      */
     if (renderTypeDeclarations) {
-      temp.track(); // tracks temp resources created to then be deleted by temp.cleanupSync
-
-      try {
-        const tmpFile = temp.openSync({ suffix: '.tsx' });
-        const tmpDir = temp.mkdirSync();
+      const compilerOptions = {
+        target,
+        module,
+        declaration: true,
+        emitDeclarationOnly: true,
+        skipLibCheck: true,
+      };
 
-        fs.writeFileSync(tmpFile.path, code);
+      const fsMap = createDefaultMapFromNodeModules({ target }, ts);
```

Comment:
Is there a reason not to pass all of `compilerOptions`?

Reply:
No, should be fine to add. I had restricted while testing, but it should be fine having read through the impl.

Reply:
Updated in the latest rev, and tests look good w/ the update.